javascript/javascript3/week3/ShoppingCart/main.js

```diff
@@ -0,0 +1,121 @@
+const productUl = document.getElementById("ul");
+const shoppingBasket = document.querySelector(".Shopping-basket");
+const selectElement = document.getElementById("product-options");
+const addButton = document.getElementById("add-to-cart");
+const totalPrice = document.getElementById("total");
+const currencyInput = document.getElementById("currency");
+
+class Product {
+    constructor(name, price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    convertToCurrency(currency) {
+        console.log(currency);
+        if (currency === "EUR") {
+            return this.price * 0.13;
+        } else if (currency === "DKK") {
+            return this.price * 1;
+        }
+    }
+}
+
+class ShoppingCart {
+    constructor() {
+        this.products = [];
+    }
+
+    addProduct(product) {
+        console.log(product);
+        this.products.push(product);
+    }
+
+    removeProduct(product) {
+        this.products = this.products.filter(item => item !== product);
+    }
+
+    searchProduct(productName) {
+        productName = productName.trim().toLocaleLowerCase();
+        return this.products.filter(product => product.name === productName);
+    }
+
+    getTotal(selectedCurrency) {
+        return this.products.reduce((accumulator, product) => accumulator + product.convertToCurrency(selectedCurrency), 0);
+    }
+
+    renderProducts(selectedCurrency) {
+
+        productUl.innerHTML = "";
+        shoppingBasket.style.visibility = "visible";
+        this.products.forEach(product => {
+            const productLi = document.createElement("li");
+            productLi.classList.add("list-item");
+            productLi.innerHTML = `Product name: ${product.name}<br/>Price: ${product.convertToCurrency(selectedCurrency)}`;
+            totalPrice.innerHTML = `Total price: ${this.getTotal(selectedCurrency)}`;
+
+            const removeButton = document.createElement("button");
+            removeButton.innerText = "Remove";
+            productLi.appendChild(removeButton);
+            productUl.appendChild(productLi);
+            removeButton.addEventListener("click", () => {
+                this.removeProduct(product);
+                totalPrice.innerHTML = `Total price: ${this.getTotal(selectedCurrency)}`;
+                this.renderProducts(selectedCurrency);
+            })
+        })
+
+    }
+
+    getUser() {
+        fetch(`https://jsonplaceholder.typicode.com/users/1`)
```

Comment:
user `1` is hard-coded. It should be passed in as a parameter to the function and substituted here in the fetch.